src/common/common.ts

```diff
@@ -396,6 +404,361 @@ export async function getVerifiableCredentialApiConfiguration(
   };
 }
 
+/**
+ * @param response Takes a response from a VC service and checks that it has the correct status and content type
+ * @param vcUrl The URL of the VC
+ * @returns The input response
+ */
+function validateVcResponse(response: Response, vcUrl: string): Response {
+  if (!response.ok) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: ${response.status} ${response.statusText}`
+    );
+  }
+
+  const contentType = response.headers.get("Content-Type");
+  if (!contentType) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: Response does not have a Content-Type header; expected application/ld+json`
+    );
+  }
+
+  const parsedContentType = contentTypeParser.parse(contentType);
+  const [mediaType, subtypesString] = parsedContentType.type.split("/");
+  const subtypes = subtypesString.split("+");
+
+  if (
+    mediaType !== "application" ||
+    // TODO: See if the response is expected to include the ld
+    // || !subtypes.includes('ld')
+    !subtypes.includes("json")
```

Comment:
I have added additional validation of the content-type here (previously this library was accepting any content type).

Currently it is accepting `application/json`, and anything of the form `application/x+json` including `application/ld+json`.

This comment is asking whether the `ld` should ever be required when it comes to VCs.

Reply:
I scanned the VC data model and the VC API specs quickly, and didn't find references to `application/ld+json` being required, so I would be on the flexible side there and not require the `ld` part.